auslib/test/web/test_client.py

```diff
@@ -854,3 +870,19 @@ def testSentryRealError(self):
             with mock.patch("auslib.web.base.sentry") as sentry:
                 self.client.get("/update/4/b/1.0/1/p/l/a/a/a/a/1/update.xml")
                 self.assertTrue(sentry.captureException.called)
+
+    def testNonSubstitutedUrlVariablesReturnEmptyUpdate(self):
+        request1 = '/update/1/%PRODUCT%/%VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/update.xml'
+        request2 = '/update/2/%PRODUCT%/%VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/update.xml'
+        request3 = '/update/3/%PRODUCT%/%VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/' \
+            'update.xml'
+        request4 = '/update/4/%PRODUCT%/%VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/' \
+            '%MOZ_VERSION%/update.xml'
+        request5 = '/update/5/%PRODUCT%/%VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/' \
+            '%IMEI%/update.xml'
+        request6 = '/update/6/%PRODUCT%/%VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%SYSTEM_CAPABILITIES%/%DISTRIBUTION%/' \
+            '%DISTRIBUTION_VERSION%/update.xml'
+
+        for request in [request1, request2, request3, request4, request5, request6]:
+            ret = self.client.get(request)
+            self.assertUpdatesAreEmpty(ret)
```

Comment:
I'm sorry, I totally missed those! You're right - you've got enough test coverage.